File: 15/navs.rb
#encoding: utf-8
require 'selenium-webdriver'

dr = Selenium::WebDriver.for :chrome
file_path = 'file:///' + File.expand_path(File.join('.', 'navs.html'))
dr.get file_path

# 方法1：层级定位，先定位ul再定位li
dr.find_element(:class, 'nav').find_element(:link_text, 'About').click()
sleep(1)

# 方法2: 直接定位link
dr.find_element(:link_text, 'Home').click()

dr.quit()

File: 15/Navs.java
import java.io.File;
import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;


public class Navs {

	public static void main(String[] args) throws InterruptedException {
		WebDriver dr = new ChromeDriver();
		
		File file = new File("src/navs.html");
		String filePath = "file:///" + file.getAbsolutePath();
		System.out.printf("now accesss %s \n", filePath);
		
		dr.get(filePath);
		Thread.sleep(1000);
		
//		方法1：层级定位，先定位ul再定位li
		dr.findElement(By.className("nav")).findElement(By.linkText("About")).click();
		Thread.sleep(1000);
		
//		方法2: 直接定位link
		dr.findElement(By.linkText("Home")).click();
		
		Thread.sleep(1000);
		System.out.println("browser will be close");
		dr.quit();	
	}

}
